internal/controllers/machine/machine_controller_noderef.go

```diff
@@ -269,16 +240,48 @@ func (r *Reconciler) getNode(ctx context.Context, c client.Reader, providerID *n
 	return &nodeList.Items[0], nil
 }
 
-func (r *Reconciler) reconcileNodeTaints(ctx context.Context, remoteClient client.Client, node *corev1.Node) error {
-	patchHelper, err := patch.NewHelper(node, remoteClient)
-	if err != nil {
-		return errors.Wrapf(err, "failed to create patch helper for Node %s", klog.KObj(node))
+// PatchNode is required to workaround an issue on Node.Status.Address which is incorrectly annotated as patchStrategy=merge
+// and this causes SSA patch to fail in case there are two address with the same key https://github.com/kubernetes-sigs/cluster-api/issues/8417
```

Comment:
```suggestion
// PatchNode is required to workaround an issue on Node.Status.Address which is incorrectly annotated as patchStrategy=merge
// and this causes SSA patch to fail in case there are two addresses with the same key https://github.com/kubernetes-sigs/cluster-api/issues/8417
```